src/gpu/intel/ocl/reusable_softmax.cl

```diff
@@ -72,3 +74,140 @@ reusable_softmax_fwd_generic(__global SRC_DATA_T *src, __global DST_DATA_T *dst,
         dst[c - begin] = TO_DST(unscaled * scale);
     }
 }
+
+#endif
+
+#ifdef USE_VECTORIZED_KERNEL
+
+__attribute__((reqd_work_group_size(SUBGROUP_SIZE, 1, 1)))
+__attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) __kernel void
+reusable_softmax_fwd_generic(__global DATA_T *src, __global DST_DATA_T *dst,
+        __global float *src_scale, __global float *dst_scale,
+        dim_t SOFTMAX_AXIS_SIZE, dim_t softmax_axis_stride,
+        dim_t softmax_axis_chunk_size, dispatch_gws_rt_params_t gws_params) {
+
+    float scale = 1.0f;
+
+    const off_t linear_thread_id = get_global_id(0);
+    const off_t data_off
+            = (linear_thread_id / SUBGROUP_SIZE) * SOFTMAX_AXIS_SIZE;
+    global DATA_T *src_backup = src;
+    src += data_off;
+
+    VECT_FLOAT_T dk;
+    float max_ = -INFINITY;
+    float denom_ = 0.0f;
+    const bool has_tail = SOFTMAX_AXIS_SIZE % (SUBGROUP_SIZE * 8);
+    int last_buf = div_up(SOFTMAX_AXIS_SIZE, (SUBGROUP_SIZE * 8));
+    if (has_tail) last_buf--;
+
+    for (int k = 0; k < last_buf; ++k) {
+        off_t idx = k * SUBGROUP_SIZE;
```

Comment:
The `last_buf` and the corresponding loops should use `off_t` as well.

Another minor comment: it may make sense to just remove `k` here and do:
```C++
off_t idx_end = SOFTMAX_AXIS_SIZE - (SUB_GROUP_SIZE * 8 + 1)
for(off_t idx = 0; idx < idx_end; idx += SUB_GROUP_SIZE * 8) {
...
```
By doing this, we could support arbitrary inner-loop sizes for minimal cost as add/subtract are much faster than mul/div operations.